### DISCUSSION TITLE
TileMap Improvement

### DISCUSSION THREAD

**Discussion by GiuseppeZani** [godotengine/godot-proposals#5691] [2D]
### Describe the project you are working on

2d

### Describe the problem or limitation you are having in your project

Hallo have some suggestions for the TileMap node.
Thanks a lot for taking them into considerations.

1 - Allow to have more than one atlas in a single tile map
2 - In the editor on the bottom of the screen. shorcuts for new single tile, new autotile, new atlas button.
3 - In the editor on the bottom of the screen Multilple single tiles from selection.
4 - In the scene when Tilemap tools selected
i would like to have new single tile, new autotile, new atlas, i know is redundant, but
would be helpful for certain task, autosave
but with ctrl+z enabled (limited undo of course).
5 - I would love to have rotate tools and flip tool enabled for current selection in the scene.

Thanks all for your amazing job.

░░░░░░░░░░░░░░░░░░░░░░█████████
░░███████░░░░░░░░░░███▒▒▒▒▒▒▒▒███
░░█▒▒▒▒▒▒█░░░░░░░███▒▒▒▒▒▒▒▒▒▒▒▒▒███
░░░█▒▒▒▒▒▒█░░░░██▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒██
░░░░█▒▒▒▒▒█░░░██▒▒▒▒▒██▒▒▒▒▒▒██▒▒▒▒▒███
░░░░░█▒▒▒█░░░█▒▒▒▒▒▒████▒▒▒▒████▒▒▒▒▒▒██
░░░█████████████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒██
░░░█▒▒▒▒▒▒▒▒▒▒▒▒█▒▒▒▒▒▒▒▒▒█▒▒▒▒▒▒▒▒▒▒▒██
░██▒▒▒▒▒▒▒▒▒▒▒▒▒█▒▒▒██▒▒▒▒▒▒▒▒▒▒██▒▒▒▒██
██▒▒▒███████████▒▒▒▒▒██▒▒▒▒▒▒▒▒██▒▒▒▒▒██
█▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒█▒▒▒▒▒▒████████▒▒▒▒▒▒▒██
██▒▒▒▒▒▒▒▒▒▒▒▒▒▒█▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒██
░█▒▒▒███████████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒██
░██▒▒▒▒▒▒▒▒▒▒████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒█
░░████████████░░░█████████████████

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Hallo have some suggestions for the TileMap node.
Thanks a lot for taking them into considerations.

1 - Allow to have more than one atlas in a single tile map
2 - In the editor on the bottom of the screen. shorcuts for new single tile, new autotile, new atlas button.
3 - In the editor on the bottom of the screen Multilple single tiles from selection.
4 - In the scene when Tilemap tools selected
i would like to have new single tile, new autotile, new atlas, i know is redundant, but
would be helpful for certain task, autosave
but with ctrl+z enabled (limited undo of course).
5 - I would love to have rotate tools and flip tool enabled for current selection in the scene.

Thanks all for your amazing job.

░░░░░░░░░░░░░░░░░░░░░░█████████
░░███████░░░░░░░░░░███▒▒▒▒▒▒▒▒███
░░█▒▒▒▒▒▒█░░░░░░░███▒▒▒▒▒▒▒▒▒▒▒▒▒███
░░░█▒▒▒▒▒▒█░░░░██▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒██
░░░░█▒▒▒▒▒█░░░██▒▒▒▒▒██▒▒▒▒▒▒██▒▒▒▒▒███
░░░░░█▒▒▒█░░░█▒▒▒▒▒▒████▒▒▒▒████▒▒▒▒▒▒██
░░░█████████████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒██
░░░█▒▒▒▒▒▒▒▒▒▒▒▒█▒▒▒▒▒▒▒▒▒█▒▒▒▒▒▒▒▒▒▒▒██
░██▒▒▒▒▒▒▒▒▒▒▒▒▒█▒▒▒██▒▒▒▒▒▒▒▒▒▒██▒▒▒▒██
██▒▒▒███████████▒▒▒▒▒██▒▒▒▒▒▒▒▒██▒▒▒▒▒██
█▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒█▒▒▒▒▒▒████████▒▒▒▒▒▒▒██
██▒▒▒▒▒▒▒▒▒▒▒▒▒▒█▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒██
░█▒▒▒███████████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒██
░██▒▒▒▒▒▒▒▒▒▒████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒█
░░████████████░░░█████████████████

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

No

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No

### Is there a reason why this should be core and not an add-on in the asset library?

Why it's the TileMap Editor

**Comment by KoBeWi** [comment]
Have you tried the new tile editor in Godot 4.0? Some of the points you listed are already implemented (and some are no longer relevant).
